packages/@aws-cdk/pipelines/lib/actions/update-pipeline-action.ts

```diff
@@ -97,7 +98,12 @@ export class UpdatePipelineAction extends CoreConstruct implements codepipeline.
     // allow the self-mutating project permissions to assume the bootstrap Action role
     selfMutationProject.addToRolePolicy(new iam.PolicyStatement({
       actions: ['sts:AssumeRole'],
-      resources: ['arn:*:iam::*:role/*-deploy-role-*', 'arn:*:iam::*:role/*-publishing-role-*'],
+      resources: [`arn:*:iam::${Stack.of(this).account}:role/*`],
```

Comment:
@njlynch we were taking benefit of the fact that the self mutating CodeBuild project has permissions to assume roles in ANY account with `arn:*:iam::*:role/` . We used this for deploying some prerequisites before the actual `application stage` to deploy something cross-account.

Now the `assumeRole` was restricted to `arn:*:iam::${Stack.of(this).account}:role/`...I know our approach might have been working by mistake, but if not, is it possible this restriction to be reverted back to `arn:*:iam::*:role/`?

@rix0rrr what are your thoughts?

Reply:
We are not going to revert this, but if you want to add additional permissions to assume roles, you can use the construct tree to dig out the relevant CodeBuild Project construct and call `addToRolePolicy()` on it for the roles that you need.